src/main/java/com/spring/onlinebookstore/service/UserServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package com.spring.onlinebookstore.service;
+
+import com.spring.onlinebookstore.dto.user.UserRegistrationRequestDto;
+import com.spring.onlinebookstore.dto.user.UserResponse;
+import com.spring.onlinebookstore.exception.RegistrationException;
+import com.spring.onlinebookstore.mapper.UserMapper;
+import com.spring.onlinebookstore.model.User;
+import com.spring.onlinebookstore.repository.user.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponse register(UserRegistrationRequestDto userRegistrationRequestDto)
+            throws RegistrationException {
+        if (userRepository.findUserByEmail(userRegistrationRequestDto.email()).isPresent()) {
```

Comment:
```suggestion
        if (userRepository.existsByEmail(userRegistrationRequestDto.email())) {
```

it's better to avoid calling isPresent()